### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by jewertow**
## Changes
- Remove usages of deprecated `ActorMaterializer()`
- Use `Source.futureSource` instead of deprecated `Source.fromFutureSource`
- Replace deprecated `schedule` with `scheduleWithFixedDelay` in `PubSubThrottler`
- Replace deprecated `schedulePeriodically` with `scheduleWithFixedDelay` in `EventsByTagStage`

### CODE REVIEW

File: core/src/main/scala/akka/persistence/cassandra/EventsByTagMigration.scala [hunk 1]
```diff
@@ -100,7 +100,7 @@ object EventsByTagMigration {
 class EventsByTagMigration(system: ActorSystem, pluginConfigPath: String = "akka.persistence.cassandra") {
   private[akka] val log = Logging.getLogger(system, getClass)
   private lazy val queries = PersistenceQuery(system).readJournalFor[CassandraReadJournal](pluginConfigPath + ".query")
-  private implicit val materialiser = ActorMaterializer()(system)
+  private implicit val sys: ActorSystem = system
```
Comment (jewertow) [inline core/src/main/scala/akka/persistence/cassandra/EventsByTagMigration.scala]: Implicit materializer is provided if there is an implicit `ActorSystem` available, so manually created materializers were replaced with implicitly defined actor systems.

Comment (chbatey) [reply core/src/main/scala/akka/persistence/cassandra/EventsByTagMigration.scala]: 👍

File: core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala [hunk 2]
```diff
@@ -64,7 +63,7 @@ class CassandraJournal(cfg: Config, cfgPath: String) extends AsyncWriteJournal w
   private val log: LoggingAdapter = Logging(context.system, getClass)
 
   private implicit val ec: ExecutionContext = context.dispatcher
-  private implicit val materializer: ActorMaterializer = ActorMaterializer()(context.system)
+  private implicit val sys: ActorSystem = context.system
```
Comment (chbatey) [inline core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala]: Can just do import context.system as it is already implicit i think

File: core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala [hunk 3]
```diff
@@ -606,12 +606,17 @@ class CassandraReadJournal(system: ExtendedActorSystem, cfg: Config, cfgPath: St
...
+        persistentRepr.persistenceId,
+        persistentRepr.sequenceNr,
+        payload,
+        persistentRepr.timestamp)
     }
 
   private[this] def toEventEnvelope(persistentRepr: PersistentRepr, offset: Offset): immutable.Iterable[EventEnvelope] =
     adaptFromJournal(persistentRepr).map { payload =>
-      EventEnvelope(offset, persistentRepr.persistenceId, persistentRepr.sequenceNr, payload)
+      EventEnvelope(offset, persistentRepr.persistenceId, persistentRepr.sequenceNr, payload, persistentRepr.timestamp)
```
Comment (chbatey) [inline core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: Revert the timestamp ones for now as it is the timestamp in the offset we want which is fixed in https://github.com/akka/akka-persistence-cassandra/pull/669/files

File: core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala [hunk 4]
```diff
@@ -606,12 +606,17 @@ class CassandraReadJournal(system: ExtendedActorSystem, cfg: Config, cfgPath: St
 
   private[this] def toEventEnvelopes(persistentRepr: PersistentRepr, offset: Long): immutable.Iterable[EventEnvelope] =
     adaptFromJournal(persistentRepr).map { payload =>
-      EventEnvelope(Offset.sequence(offset), persistentRepr.persistenceId, persistentRepr.sequenceNr, payload)
+      EventEnvelope(
+        Offset.sequence(offset),
+        persistentRepr.persistenceId,
+        persistentRepr.sequenceNr,
+        payload,
+        persistentRepr.timestamp)
```
Comment (chbatey) [inline core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: And revert this one

File: core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala (hunk 3 above)
Comment (jewertow) [reply core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: Done

File: core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala (hunk 4 above)
Comment (jewertow) [reply core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: Done

File: core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala (hunk 2 above)
Comment (jewertow) [reply core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala]: Done

File: core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala [hunk 5]
```diff
@@ -421,7 +421,7 @@ import scala.compat.java8.FutureConverters._
         }
 
         def optionallySchedule(key: Any, duration: Option[FiniteDuration]): Unit = {
-          duration.foreach(fd => schedulePeriodically(key, fd))
+          duration.foreach(fd => scheduleWithFixedDelay(key, FiniteDuration(0L, SECONDS), fd))
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala]: to keep same initial interval as before this should be `fd, fd`

File: core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala [hunk 6]
```diff
@@ -120,7 +120,7 @@ class CassandraReadJournal(system: ExtendedActorSystem, cfg: Config, cfgPath: St
   private val serialization = SerializationExtension(system)
   implicit private val ec =
     system.dispatchers.lookup(querySettings.pluginDispatcher)
-  implicit private val materializer = ActorMaterializer()(system)
+  implicit private val sys: ActorSystem = system
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: It's somewhat wrong that we didn't use the pluginDispatcher as the dispatcher for the Materializer previously.
To honor that setting I think it's actually better to create a separate materializer here.

File: core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala [hunk 7]
```diff
@@ -72,8 +72,6 @@ import akka.annotation.InternalApi
   private def preparedSelectSnapshotMetadataWithMaxLoadAttemptsLimit: Future[PreparedStatement] =
     session.prepare(selectSnapshotMetadata(limit = Some(snapshotSettings.maxLoadAttempts)))
 
-  private implicit val materializer = ActorMaterializer()
```
Comment (patriknw) [inline core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala]: same here, better to have a materializer with the right dispatcher

File: session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSession.scala [hunk 8]
```diff
@@ -56,7 +55,7 @@ final class CassandraSession(
     extends NoSerializationVerificationNeeded {
 
   implicit private[akka] val ec = executionContext
-  private lazy implicit val materializer = ActorMaterializer()(system)
+  private implicit val sys: ActorSystem = system
```
Comment (patriknw) [inline session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSession.scala]: same here, better to have a materializer with the right dispatcher

File: core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala (hunk 5 above)
Comment (jewertow) [reply core/src/main/scala/akka/persistence/cassandra/query/EventsByTagStage.scala]: Fixed

File: core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala (hunk 6 above)
Comment (jewertow) [reply core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: For now akka-stream provides two factories for materializers:
- `def createMaterializer(systemProvider: ClassicActorSystemProvider): Materializer`
- `def createMaterializer(contextProvider: ClassicActorContextProvider): Materializer`
so I don't know how to create the materializer with above dispatcher.
Could you explain me how to do it?

Comment (patriknw) [reply core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: Sorry, I was wrong. Defining a dispatcher doesn't require a separate materializer. It's an attribute on the stream (Source or RunnableGraph). I don't know if we already set that here, otherwise we should. Then the implicit ActorSystem can be used.

```
.withAttributes(ActorAttributes.dispatcher(querySettings.pluginDispatcher)
```

Comment (jewertow) [reply core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: It seems that the attribute is set.

Comment (patriknw) [reply core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala]: good, sorry for the noise then

File: session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSession.scala (hunk 8 above)
Comment (patriknw) [inline session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSession.scala]: @jewertow could you revert this change to the `session` project? As @ennru said, we are anyway moving it to Alpakka and this would introduce conflicts with that work.
Otherwise this is ready to be merged.

Comment (jewertow) [reply session/src/main/scala/akka/cassandra/session/scaladsl/CassandraSession.scala]: Done

**Comment by patriknw** [comment]
@jewertow sorry for the inconvenience. Some conflicts now, probably just renamed package for CassandraSession

**Comment by jewertow** [comment]
@patriknw  I resolved conflicts.
